proto/babylon/finality/v1/query.proto

```diff
@@ -18,6 +18,27 @@ service Query {
     option (google.api.http).get = "/babylon/finality/v1/params";
   }
 
+  // ActiveFinalityProvidersAtHeight queries finality providers with non zero voting power at given height.
+  rpc ActiveFinalityProvidersAtHeight(QueryActiveFinalityProvidersAtHeightRequest) returns (QueryActiveFinalityProvidersAtHeightResponse) {
+    option (google.api.http).get = "/babylon/finality/v1/finality_providers/{height}";
+  }
+
+  // FinalityProviderPowerAtHeight queries the voting power of a finality provider at a given height
+  rpc FinalityProviderPowerAtHeight(QueryFinalityProviderPowerAtHeightRequest) returns (QueryFinalityProviderPowerAtHeightResponse) {
+    option (google.api.http).get = "/babylon/finality/v1/finality_providers/{fp_btc_pk_hex}/power/{height}";
+  }
+
+  // FinalityProviderCurrentPower queries the voting power of a finality provider at the current height
+  rpc FinalityProviderCurrentPower(QueryFinalityProviderCurrentPowerRequest) returns (QueryFinalityProviderCurrentPowerResponse) {
+    option (google.api.http).get = "/babylon/finality/v1/finality_providers/{fp_btc_pk_hex}/power";
+  }
+
+  // ActivatedHeight queries the height when BTC staking protocol is activated, i.e., the first height when
+  // there exists 1 finality provider with voting power
+  rpc ActivatedHeight(QueryActivatedHeightRequest) returns (QueryActivatedHeightResponse) {
+    option (google.api.http).get = "/babylon/finality/v1/activated_height";
+  }
```

Comment:
Is this query still useful given that we have activation height in the parameter?

Reply:
I think we still need it, although we might want to change its definition now. At this moment the activated height should be something like `max(BTC activation height, finality activation height)`, rather than just the BTC activation height. Will create a Github issue for this

Reply:
https://github.com/babylonlabs-io/babylon/issues/231